checks/raw/packaging.go

```diff
@@ -114,6 +118,41 @@ func Packaging(c *checker.CheckRequest) (checker.PackagingData, error) {
 	return data, nil
 }
 
+func isGoProject(c *checker.CheckRequest, data *checker.PackagingData) (bool, error) {
+	matchedFiles, err := c.RepoClient.ListFiles(func(string) (bool, error) { return true, nil })
+	if err != nil {
+		return false, fmt.Errorf("%w", err)
+	}
+	if err != nil {
+		return false, fmt.Errorf("RepoClient.ListFiles: %w", err)
+	}
+
+	foundGoSum := false
+	foundGoMod := false
+	for _, fp := range matchedFiles {
+		// Only look in the root directory.
+		if fp == "go.mod" {
+			foundGoMod = true
+		}
+		if fp == "go.sum" {
+			foundGoSum = true
+		}
+		if foundGoSum && foundGoMod {
+			pkg := checker.Package{
+				Ecosystem: ecosystemAsPointer(checker.PackageEcosystemGo),
+				Name:      stringPointer(c.Repo.URI()),
+			}
+			data.Packages = append(data.Packages, pkg)
```

Comment:
should this be in the `if ok` block of the caller?
I wouldn't expect `isGoProject` to need/modify data.

This could also be generalized to `identifyEcosystem` or something, if you're worried about too much logic in `func Packaging`

Reply:
> should this be in the if ok block of the caller?

works for me. Thanks

> This could also be generalized to identifyEcosystem or something, if you're worried about too much logic in func Packaging

We can maybe do that as part of https://github.com/ossf/scorecard/issues/2494